pacman/operations/rigged_algorithms/hilbert_placer.py

```diff
@@ -0,0 +1,171 @@
+            state.y_pos + state.change_y
+        yield state.x_pos, state.y_pos
+
+        # Turn right
+        state.change_x, state.change_y = state.change_y * angle, \
+            state.change_x * -angle
+
+        # Recurse positive
+        for state.x_pos, state.y_pos in self._hilbert_curve(level - 1,
+                                                            angle, state):
+            yield state.x_pos, state.y_pos
+
+        # Move forward
+        state.x_pos, state.y_pos = state.x_pos + state.change_x, \
+            state.y_pos + state.change_y
+        yield state.x_pos, state.y_pos
+
+        # Recurse positive
+        for state.x_pos, state.y_pos in self._hilbert_curve(level - 1,
+                                                            angle, state):
+            yield state.x_pos, state.y_pos
+
+        # Turn right
+        state.change_x, state.change_y = state.change_y * angle, \
+            state.change_x * -angle
+
+        # Move forward
+        state.x_pos, state.y_pos = state.x_pos + state.change_x, \
+            state.y_pos + state.change_y
+        yield state.x_pos, state.y_pos
+
+        # Recurse negative
+        for state.x_pos, state.y_pos in self._hilbert_curve(level - 1,
+                                                            -angle, state):
+            yield state.x_pos, state.y_pos
+
+        # Turn left
+        state.change_x, state.change_y = state.change_y * -angle, \
+            state.change_x * angle
+
+    def _generate_hilbert_chips(self, machine):
+        """A generator which iterates over a set of chips in a machine in
+        a hilbert path.
+
+        For use as a chip ordering for the sequential placer.
+        """
+        max_dimen = max(machine.max_chip_x, machine.max_chip_y)
+        hilbert_levels = int(ceil(log(max_dimen, 2.0))) if max_dimen >= 1 \
+            else 0
+        for x, y in self._hilbert_curve(hilbert_levels):
+            if machine.is_chip_at(x, y):
+                yield x, y
+
+    def _place_vertex(self, vertex, resource_tracker, machine, placements,
+                      vertices_on_same_chip):
+
+        vertices = vertices_on_same_chip[vertex]
+        chips = self._generate_hilbert_chips(machine)
+
+        if len(vertices) > 1:
+            assigned_values = \
+                resource_tracker.allocate_constrained_group_resources([
+                    (vert.resources_required, vert.constraints)
+                    for vert in vertices
+                ], chips)
+            for (x, y, p, _, _), vert in zip(assigned_values, vertices):
+                placement = Placement(vert, x, y, p)
+                placements.add_placement(placement)
+        else:
+            (x, y, p, _, _) = resource_tracker.allocate_constrained_resources(
+                vertex.resources_required, vertex.constraints, chips)
+            placement = Placement(vertex, x, y, p)
+            placements.add_placement(placement)
+
+        return vertices
```

Comment:
Moved down hilbert_curve; others are in order (ie _generate_hilbert_chips is called before _place_vertex)